core/dive.c

```diff
@@ -1657,9 +1657,10 @@ static void sample_renumber(struct sample *s, int i, const int mapping[])
 	int j;
 
 	for (j = 0; j < MAX_SENSORS; j++) {
-		int sensor;
+		int sensor = -1;
```

Comment:
Ugh. Not sure I like this 
- use an array of `int`s to map `uint8`s and have `-1` as special value (existing code)
- use `uint8_t 0xff` as a different special value
- risk that some unintended cast somewhere makes those two values the same -- or is this intentional?
My head hurts

Reply:
Yes, since `uint8_t` doesn't support `-1`, `0xff` seems like the next best choice as an in-band signal. What do you suggest?

Reply:
Yeah, much as it surprises me at times - I do realize that `uint8_t` is unsigned and as such doesn't support negative values.

Snarky comments aside, my point was that we still have signed/unsigned mixups in our code - and that something like this makes me worried about `char x = uint8_t y` and now our `0xff` has turned in to `-1`

But I guess I'm just overthinking this.
And since Linus appears busy with other things, I'll just move forward and merge what you have.

Reply:
> and that something like this makes me worried about char x = uint8_t y and now our 0xff has turned in to -1

Turning `(uint8_t)0xff` into `(int)-1` would be the good thing (invalid to invalid). The problem with unsigned types is that `(uint8_t)0xff` is cast to `(int)255`, which represents a valid cylinder index.

I don't think that we use `char` as an index variable anywhere...?